### DISCUSSION TITLE
How can I load a webpage at a specifc element using id's? Similar to using #id at the end of a url.

### DISCUSSION THREAD

**Discussion by Gome510** [react-native-webview/react-native-webview#3593] [Q&A]
When you have a link like url.com#title_id, url.com will open and scroll to the element with the id "title_id". I get an error (-1100 NSURLErrorDomain) when I attempt this. The only work around I've found is to inject javascript to use scrollIntoView(). However that has resulted in it scrolling to the wrong position. Links within the page scroll to the correct position.

**Comment by lloydchang** [comment]
**Question:**

Is your JavaScript waiting for DOM content to be fully loaded?

**Problem:**

* The `NSURLErrorDomain` `-1100` error occurs because WKWebView on iOS doesn't properly handle URL fragments (#) for scrolling to elements.

**Solution:**

* Instead of relying on native URL fragment handling, we need to manually manage the scroll behavior using JavaScript. The script needs to:

1. Extract the fragment from URL
2. Wait for DOM content to be fully loaded
3. Find the target element
4. Execute smooth scrolling with proper positioning

```JavaScript
const handleFragmentScroll = `
  setTimeout(() => {
    const id = window.location.hash.slice(1);
    document.getElementById(id)?.scrollIntoView({
      behavior: 'smooth',
      block: 'start'
    });
  }, 300);
`;

<WebView
  source={{ uri: url }}
  injectedJavaScript={handleFragmentScroll}
/>
```

This approach bypasses the iOS WKWebView limitation by implementing custom fragment navigation through JavaScript injection.
